src/com/company/Shop.java

```diff
@@ -53,19 +86,74 @@ public void fire(Employee employee) {
         employees.remove(employee);
     }
 
-    public void addProduct(ProductDetails product, double quantity) {
+
+    public void addProduct(Product product, double quantity) {
+        boolean isFound = false;
```

Comment:
looks good, but could be even better :D think how to use `findProductById`  here